docs/source/reference.rst

```diff
@@ -355,6 +361,156 @@ section with a list of pip requirement specifiers.
+list of packages in ``anaconda-project-lock.yml``, it could look
+like this for example:
+
+.. code-block:: yaml
+
+    locking_enabled: true
+
+    env_specs:
+      default:
+        locked: true
+        env_spec_hash: eb23ad7bd050fb6383fcb71958ff03db074b0525
+        platforms:
+        - all
+        packages:
+          all:
+          - backports=1.0=py27_0
+          - backports_abc=0.5=py27_0
+          - bokeh=0.12.4=py27_0
+          - futures=3.0.5=py27_0
+          - jinja2=2.9.5=py27_0
+          - markupsafe=0.23=py27_2
+          - mkl=2017.0.1=0
+          - numpy=1.12.1=py27_0
+          - pandas=0.19.2=np112py27_1
+          - pip=9.0.1=py27_1
+          - python-dateutil=2.6.0=py27_0
+          - python=2.7.13=0
+          - pytz=2016.10=py27_0
+          - pyyaml=3.12=py27_0
+          - requests=2.13.0=py27_0
+          - singledispatch=3.4.0.3=py27_0
+          - six=1.10.0=py27_0
+          - ssl_match_hostname=3.4.0.2=py27_1
+          - tornado=4.4.2=py27_0
+          - wheel=0.29.0=py27_0
+          unix:
+          - openssl=1.0.2k=1
+          - readline=6.2=2
+          - setuptools=27.2.0=py27_0
+          - sqlite=3.13.0=0
+          - tk=8.5.18=0
+          - yaml=0.1.6=0
+          - zlib=1.2.8=3
+          win:
+          - setuptools=27.2.0=py27_1
+          - vs2008_runtime=9.00.30729.5054=0
+
+By locking your versions, you can make your project more portable;
+when you share it with someone else or deploy it on a server or
+try to use it yourself in a few months, you'll get the same
+package versions you've already used for testing. If you don't
+lock your versions, you may find that your project stops working
+due to changes in its dependencies.
+
+When you're ready to test the latest versions of your
+dependencies, run ``anaconda-project update`` to update the
+versions in ``anaconda-project-lock.yml`` to the latest available.
+
+If you check ``anaconda-project-lock.yml`` into revision control
+(such as git), when you check out old versions of your project
+you'll also get the dependencies those versions were tested with.
+And you'll be able to see changes in your dependencies over time
+in your revision control history.
+
+Specifying supported platforms
+==============================
+
+Whenever you lock or update a project, dependencies are resolved
+for all platforms a project supports. This allows you to do your
+work on Windows and deploy to Linux, for example.
+
+``anaconda-project lock`` by default will add a ``platforms:
+[all]`` line to ``anaconda-project.yml``. If you don't care about
+some of the platforms included in ``all``, you might want to
+change this line. Updates will be faster if you support fewer
+platforms. Also, some projects only work on certain platforms.
+
+The ``platforms:`` line does nothing when a project is unlocked.
+
+Useful values in ``platforms:`` include:
+
+ * ``all``: all "popular" platforms, currently ``linux-64``,
+   ``linux-32``, ``win-64``, ``win-32``, ``osx-64``
```

Comment:
Now that I read this in more detail, I'd recommend changing it to "default". It's best not to call it "all" if it doesn't include the more rare architectures like PPC and ARM. It could also be called "major" or "main" or some other name.